lib/logflare/endpoints/cache.ex

```diff
@@ -1,231 +1,37 @@
-    end
-  end
-
-  def handle_call(:query, _from, %__MODULE__{cached_result: cached_result} = state) do
-    {:reply, {:ok, cached_result}, state}
-  end
+  def child_spec(_) do
+    stats = Application.get_env(:logflare, :cache_stats, false)
 
-  def handle_call(:invalidate, _from, state) do
-    {:stop, :normal, {:ok, :stopped}, state}
-  end
-
-  def handle_info(:refresh, state) do
-    task = Tasks.async(__MODULE__, :do_query, [state])
-    tasks = [task | state.query_tasks]
-
-    running = Enum.count(tasks)
-
-    if running > 1,
-      do: Logger.warning("CacheTaskError: #{running} Endpoints.Cache tasks are running")
-
-    {:noreply, %{state | query_tasks: tasks}}
-  end
-
-  def handle_info(:shutdown, state) do
-    Logger.debug("#{__MODULE__}: shutting down cache normally")
-    {:stop, :normal, state}
-  end
-
-  def handle_info({from_task, {:ok, results}}, state) do
-    tasks = Enum.reject(state.query_tasks, &(&1.pid == from_task))
-
-    if is_reference(state.refresh_timer) do
-      Process.cancel_timer(state.refresh_timer)
-    end
-
-    timer = refresh(proactive_querying_ms(state.query))
-
-    {:noreply, %{state | cached_result: results, query_tasks: tasks, refresh_timer: timer}}
-  end
-
-  def handle_info({_from_task, {:error, _response}}, state) do
-    {:stop, :normal, state}
-  end
-
-  def handle_info({:DOWN, _ref, :process, pid, :normal}, state) do
-    tasks = Enum.reject(state.query_tasks, &(&1.pid == pid))
-
-    {:noreply, %{state | query_tasks: tasks}}
-  end
-
-  def handle_info({:DOWN, _ref, :process, _pid, reason}, state) do
-    Logger.warning("#{__MODULE__}: task exited with reason #{reason}")
-    {:stop, :normal, state}
-  end
-
-  def do_query(state) do
-    Logflare.Endpoints.run_query(state.query, state.params)
-  end
-
-  def endpoints_part(query_id, params) do
-    part = :erlang.phash2({query_id, params}, System.schedulers_online())
-    "endpoints_#{part}" |> String.to_existing_atom()
-  end
-
-  def endpoints_part(query_id) do
-    part = :erlang.phash2(query_id, System.schedulers_online())
-    "endpoints_#{part}" |> String.to_existing_atom()
-  end
-
-  defp refresh(every) do
-    Process.send_after(self(), :refresh, every)
-  end
-
-  defp shutdown(every) do
-    Process.send_after(self(), :shutdown, every)
-  end
-
-  defp proactive_querying_ms(query) do
-    query.proactive_requerying_seconds * 1_000
-  end
-
-  defp cache_duration_ms(query) do
-    query.cache_duration_seconds * 1_000
-  end
-
-  defp fetch_latest_query_endpoint(state) do
     %{
-      state
-      | query:
-          Endpoints.get_mapped_query_by_token(state.query.token)
-          |> Logflare.Repo.preload(:user)
+      id: __MODULE__,
+      start:
+        {Cachex, :start_link,
+         [
+           __MODULE__,
+           [
+             hooks:
+               [
+                 if(stats, do: Utils.cache_stats()),
+                 Utils.cache_limit(100_000)
+               ]
+               |> Enum.filter(& &1),
```

Comment:
nit: I usually find `Enum.reject(&is_nil/1)` clearer - this is completely fine of course